data/binding/preference_test.go

```diff
@@ -0,0 +1,107 @@
+package binding
+
+import (
+	"testing"
+	"time"
+
+	"fyne.io/fyne/test"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBindPreferenceBool(t *testing.T) {
+	a := test.NewApp()
+	p := a.Preferences()
+	key := "test-bool"
+
+	p.SetBool(key, true)
+	bind := BindPreferenceBool(key, p)
+	assert.True(t, bind.Get())
+
+	bind.Set(false)
+	assert.False(t, bind.Get())
+	assert.False(t, p.Bool(key))
```

Comment:
Ensure that calling `p.SetBool(key, true)` again will trigger the binding

Reply:
And ditto for other types

Reply:
Sadly it won't yet - as we don't have any preference listener (see comment in your other feedback).
We could have the properties code depend on dataBinding, and have a specific "PreferenceUpdated()" called, not sure if this is a good approach or a workaround...?

I'd be happy to open a new issue "Preferences set directly do not trigger binding changes" so we can implement it properly? I think it would also need to include code so that the "preferencesLoad" validatess which items have changed so we don't just blindly call back every preference value if the file is touched...

Reply:
Yes let's open an issue to track this, though I wouldn't worry about blindly calling back - data bindings should be smart enough to only trigger their listeners when their value changes.

Reply:
#1577